util_scripts/Builder/Program.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using EnvDTE;
+using EnvDTE80;
+using System.Threading;
+
+namespace BeckhoffBuilder
+{   
+    class VSVersion {
+        public static readonly VSVersion VS_2010 = new VSVersion("VisualStudio.DTE.10.0");
+        public static readonly VSVersion VS_2012 = new VSVersion("VisualStudio.DTE.11.0");
+        public static readonly VSVersion VS_2013 = new VSVersion("VisualStudio.DTE.12.0");
+        public static readonly VSVersion VS_2015 = new VSVersion("VisualStudio.DTE.14.0");
+        public static readonly VSVersion VS_2017 = new VSVersion("VisualStudio.DTE.15.0");
+
+        public String DTEDesc;
+
+        public VSVersion(String DTEDesc) {
+            this.DTEDesc = DTEDesc;
+        }
+    }
+
+    enum ErrorCode {
+        SUCCESS = 0, REGISTER_FAILED = 1, BUILD_FAILED = 2
+    }
+
+    class Main
+    {
+        private System.Threading.Thread thread;
+        private String slnPath;
+        public ErrorCode errorCode = 0;
+
+        /// <summary>
+        /// The main class of the program.
+        /// </summary>
+        /// <param name="path">A path to the solution to build.</param>
+        public Main(String path)
+        {
+            this.slnPath = path;
+            //Registering the message filter must be done on a STA thread.
+            thread = new System.Threading.Thread(run);
+            thread.SetApartmentState(ApartmentState.STA);
+            thread.Start();
+            thread.Join();
+        }
+
+        /// <summary>
+        /// Gets the DTE, which causes the Visual Studio process to start up in the background.
+        /// </summary>
+        /// <param name="version">The version of Visual Studio to use, specified as a <see cref="VSVersion"/>.</param>
+        /// <returns>The DTE object that can be used to interact with Visual Studio.</returns>
+        private EnvDTE80.DTE2 getDTE(VSVersion version)

```

Comment:
Can this do any more validation of the build environment to make sure that an appropriate version of visual studio and TwinCAT are installed?

Reply:
I couldn't find an easy way to get the required version out of the codebase so I'm just checking it's greater than 3.1